src/react/hooks/__tests__/useSubscription.test.tsx

```diff
@@ -1123,6 +1125,231 @@ followed by new in-flight setup", async () => {
+            restart: expect.any(Function),
+            variables: undefined,
+          });
+          expect(snapshot.error).toBeInstanceOf(ApolloError);
+        }
+
+        expect(onError).toHaveBeenCalledTimes(1);
+        expect(onError).toHaveBeenCalledWith(
+          graphQlErrorResult.result!.errors![0]
+        );
+        expect(onError).toHaveBeenCalledWith(expect.any(ApolloError));
+        expect(onData).toHaveBeenCalledTimes(0);
+        expect(errorBoundaryOnError).toHaveBeenCalledTimes(0);
+      });
+      it("`errorPolicy: 'ignore'`: returns `{ data }`, calls `onData`", async () => {
+        const {
+          ProfiledHook,
+          wrapper,
+          link,
+          graphQlErrorResult,
+          errorBoundaryOnError,
+        } = setup();
+        const onData = jest.fn();
+        const onError = jest.fn();
+        render(
+          <ProfiledHook
+            errorPolicy="ignore"
+            onError={onError}
+            onData={onData}
+          />,
+          {
+            wrapper,
+          }
+        );
+
+        await ProfiledHook.takeSnapshot();
+        link.simulateResult(graphQlErrorResult);
+        {
+          const snapshot = await ProfiledHook.takeSnapshot();
+          expect(snapshot).toStrictEqual({
+            loading: false,
+            error: undefined,
+            data: { totalLikes: 42 },
+            restart: expect.any(Function),
+            variables: undefined,
+          });
+        }
+
+        expect(onError).toHaveBeenCalledTimes(0);
+        expect(onData).toHaveBeenCalledTimes(1);
+        expect(onData).toHaveBeenCalledWith({
+          client: expect.anything(),
+          data: {
+            data: { totalLikes: 42 },
+            loading: false,
+            // should this be undefined?
+            error: undefined,
+            variables: undefined,
+          },
+        });
+        expect(errorBoundaryOnError).toHaveBeenCalledTimes(0);
+      });
+    });
+    describe("protocol error", () => {
+      it.each([undefined, "none", "all", "ignore"] as const)(
+        "`errorPolicy: '%s'`: returns `{ error }`, calls `onError`",
+        async (errorPolicy) => {
+          const {
+            ProfiledHook,
+            wrapper,
+            link,
+            protocolErrorResult,
+            errorBoundaryOnError,
+          } = setup();
+          const onData = jest.fn();
+          const onError = jest.fn();
+          render(
+            <ProfiledHook
+              errorPolicy={errorPolicy}
+              onError={onError}
+              onData={onData}
+            />,
+            {
+              wrapper,
+            }
+          );
+
+          await ProfiledHook.takeSnapshot();
+          link.simulateResult(protocolErrorResult);
+          {
+            const snapshot = await ProfiledHook.takeSnapshot();
+            expect(snapshot).toStrictEqual({
+              loading: false,
+              error: new ApolloError({ errorMessage: "test" }),
```

Comment:
Similar here. Can we check this includes protocol errors?

```suggestion
              error: new ApolloError({ 
                errorMessage: "test",
                protocolErrors: [...]
              }),
```